src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj

```diff
@@ -38,4 +38,47 @@
     <InternalsVisibleToTest Include="$(AssemblyName).Tests" />
     <InternalsVisibleToDynamicProxyGenAssembly2 Include="*" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <!-- Defines whether we need to pack buildTransitive artifacts for net462. -->
+    <PackNet462 Condition=" '$(MinimumSupportedTfmForPackaging)' != '' and
+                            '$(ConditionalNet462)' != '' and
+                            ($(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)')) and $(TargetFrameworks.Contains('$(ConditionalNet462)'))) "
+                >true</PackNet462>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="buildTransitive\$(MSBuildProjectName).*"
+          CopyToOutputDirectory="PreserveNewest"
+          Pack="true"
+          PackagePath="buildTransitive\$(MinimumSupportedTfmForPackaging)\"
+          Condition="'$(MinimumSupportedTfmForPackaging)' != '' and $(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)'))" />
+    
+    <!-- For net462 we copy all buildTransitive files except the .targets file. We have to rename the .targets file
+         when copying it because our build process automatically adds a .targets file with the same name as the project
+         file. Then we'll import the renamed .targets file in the automatically added .targets file. For that we need
+         to set the _AdditionalNETStandardCompatErrorFileContents variable. -->
+    <None Include="buildTransitive\$(MSBuildProjectName).*"
+          Exclude="buildTransitive\$(MSBuildProjectName).targets"
+          CopyToOutputDirectory="PreserveNewest"
+          Pack="true"
+          PackagePath="buildTransitive\net462\"
+          Condition="'$(PackNet462)' == 'true'" />
+    
+    <None Include="buildTransitive\$(MSBuildProjectName).targets"
+          CopyToOutputDirectory="PreserveNewest"
+          Pack="true"
+          PackagePath="buildTransitive\net462\$(MSBuildProjectName).net462.targets"
+          Condition="'$(PackNet462)' == 'true'" />
```

Comment:
Whilst this works in this current state, I don't think it's correct in general.
It's conceivable that in the future there could be other *.targets files added, but they won't get packaged.

My original proposal in https://github.com/dotnet/extensions/pull/5363#discussion_r1741290810 was generalised, and will work in any situation, i.e.:
1. Pack everything under "buildTransitive" folder.
2. For net462, if it's used, pack "buildTransitive\\Microsoft.Extensions.Http.Resilience.targets" as "buildTransitive\\Microsoft.Extensions.Http.Resilience.net462.targets" because the original "buildTransitive\\Microsoft.Extensions.Http.Resilience.targets" will get overwritten.

```suggestion
    <!--
        For net462 we have to rename the .targets file because our build process will overwrite the content
        of the original buildTransitive\Microsoft.Extensions.Http.Resilience.targets.
        We import the copy via _AdditionalNETStandardCompatErrorFileContents variable.
      -->
    <None Include="buildTransitive\Microsoft.Extensions.Http.Resilience.targets"
          Condition="'$(MinimumSupportedTfmForPackaging)' != '' and
                     '$(ConditionalNet462)' != '' and
                     ($(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)')) and $(TargetFrameworks.Contains('$(ConditionalNet462)')))"
          CopyToOutputDirectory="PreserveNewest"
          Pack="true"
          PackagePath="buildTransitive\net462\Microsoft.Extensions.Http.Resilience.net462.targets" />
```

Reply:
> I partially applied your suggestion. In particular,
> 
> * We pack everything for `MinimumSupportedTfmForPackaging`.
> * We pack everything except M.E.Http.Resilience.targets for `net462`. We have to explicitly exclude it from copying, because otherwise we get the following error, saying that msbuild cannot overwrite the file:
>   ```
>   C:\Repos\github\dotnet-extensions\.dotnet\sdk\8.0.300\Sdks\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets(221,5): warning NU5118: File 'C:\Repos\github\dotnet-extensions\artifacts\obj\Microsoft.Extensi
>   ons.Http.Resilience\netstandardcompaterror_net462.targets' is not added because the package already contains file 'buildTransitive\net462\Microsoft.Extensions.Http.Resilience.targets' [C:\Repos\github\dotnet-extensions\src\Libra 
>   ries\Microsoft.Extensions.Http.Resilience\Microsoft.Extensions.Http.Resilience.csproj]
>   ```
> * For `net462` we copy M.E.Http.Resilience.targets renaming it to M.E.Http.Resilience.net462.targets.
> 
> What do you think?

The reason you get the error is because there's a duplicated collection. There have to be only two items:
```diff
diff --git a/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj b/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
index f0499dada2..340fdb2abf 100644
--- a/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
+++ b/src/Libraries/Microsoft.Extensions.Http.Resilience/Microsoft.Extensions.Http.Resilience.csproj
@@ -41,22 +41,14 @@
 
   <ItemGroup>
     <None Include="buildTransitive\**\*"
-          CopyToOutputDirectory="PreserveNewest"
-          Pack="true"
-          PackagePath="buildTransitive\$(MinimumSupportedTfmForPackaging)\"
-          Condition="'$(MinimumSupportedTfmForPackaging)' != '' and $(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)'))" />
+           CopyToOutputDirectory="PreserveNewest"
+           Pack="true"
+           PackagePath="buildTransitive\$(MinimumSupportedTfmForPackaging)\" />
 
     <!-- For net462 we copy all buildTransitive files except the .targets file. We have to rename the .targets file
          when copying it because our build process automatically adds a .targets file with the same name as the project
          file. Then we'll import the renamed .targets file in the automatically added .targets file. For that we need
          to set the _AdditionalNETStandardCompatErrorFileContents variable. -->
-    <None Include="buildTransitive\**\*"
-          Exclude="buildTransitive\$(MSBuildProjectName).targets"
-          CopyToOutputDirectory="PreserveNewest"
-          Pack="true"
-          PackagePath="buildTransitive\net462\"
-          Condition="'$(IsPackNet462)' == 'true'" />
-    
     <None Include="buildTransitive\$(MSBuildProjectName).targets"
           CopyToOutputDirectory="PreserveNewest"
           Pack="true"
```

With that the package looks as follows:
![image](https://github.com/user-attachments/assets/6507f96f-1e58-45f3-b0f0-62a6f88a91d2)

Reply:
> The reason you get the error is because there's a duplicated collection. There have to be only two items:

Those collections are not duplicates. They copy files to different folders: net462 and net6.0. Please, note the difference:

```xml
    <None Include="buildTransitive\**\*"
          CopyToOutputDirectory="PreserveNewest"
          Pack="true"
          PackagePath="buildTransitive\$(MinimumSupportedTfmForPackaging)\"   <--- Copy to net6.0 folder.
          Condition="'$(MinimumSupportedTfmForPackaging)' != '' and $(TargetFrameworks.Contains('$(MinimumSupportedTfmForPackaging)'))" />

    <None Include="buildTransitive\**\*"
          Exclude="buildTransitive\$(MSBuildProjectName).targets"   <--- We have to exclude this file to avoid the warning.
          CopyToOutputDirectory="PreserveNewest"
          Pack="true"
          PackagePath="buildTransitive\net462\"   <--- Copy to net462 folder.
          Condition="'$(IsPackNet462)' == 'true'" />
```